libs/vertexai/langchain_google_vertexai/vectorstores/vectorstores.py

```diff
@@ -166,22 +173,42 @@ def add_texts(
         self,
         texts: Iterable[str],
         metadatas: Union[List[dict], None] = None,
+        ids: Union[List[str], None] = None,
         is_complete_overwrite: bool = False,
         **kwargs: Any,
     ) -> List[str]:
         """Run more texts through the embeddings and add to the vectorstore.
+
         Args:
             texts: Iterable of strings to add to the vectorstore.
             metadatas: Optional list of metadatas associated with the texts.
+            ids: Optional list of ids to be assigned to the texts in the index.
+                If None, unique ids will be generated.
+            is_complete_overwrite: Optional, determines whether this is an append or
+                overwrite operation. Only relevant for BATCH UPDATE indexes.
             kwargs: vectorstore specific parameters.
+
         Returns:
             List of ids from adding the texts into the vectorstore.
         """
 
-        # Makes sure is a list an can get the length, should we support iterables?
+        # Makes sure is a list and can get the length, should we support iterables?
         # metadata is a list so probably not?
         texts = list(texts)
-        ids = self._generate_unique_ids(len(texts))
+
+        if ids is None:
```

Comment:
nits: for a little bit better readability, it might be a good idea to move generating of ids after checks that ids provided are valid :).